File: sdk/src/main/java/com/cnblogs/sdk/model/UserInfo.java
package com.cnblogs.sdk.model;

/**
 * 用户信息
 * @author RAE
 * @date 2021/02/23
 */
public class UserInfo {

    /**
     * 账号
     */
    private String spaceUserId;

    /**
     * 昵称
     */
    private String displayName;

    /**
     * blogApp,非常重要的字段,唯一ID
     */
    private String blogApp;

    /**
     * 未读消息数量
     */
    private int unreadMsg;

    /**
     * 个人中心主页
     */
    private String homeLink;

    /**
     * 博客主页
     */
    private String blogLink;

    /**
     * 头像小图
     */
    private String iconName;

    /**
     * 头像大图
     */
    private String avatarName;

    public UserInfo() {
    }

    public String getSpaceUserId() {
        return spaceUserId;
    }

    public void setSpaceUserId(String spaceUserId) {
        this.spaceUserId = spaceUserId;
    }

    public String getDisplayName() {
        return displayName;
    }

    public void setDisplayName(String displayName) {
        this.displayName = displayName;
    }

    public String getBlogApp() {
        if (blogApp == null) {
            return this.spaceUserId;
        }
        return blogApp;
    }

    public void setBlogApp(String blogApp) {
        this.blogApp = blogApp;
    }

    public int getUnreadMsg() {
        return unreadMsg;
    }

    public void setUnreadMsg(int unreadMsg) {
        this.unreadMsg = unreadMsg;
    }

    public String getHomeLink() {
        return homeLink;
    }

    public void setHomeLink(String homeLink) {
        this.homeLink = homeLink;
    }

    public String getBlogLink() {
        return blogLink;
    }

    public void setBlogLink(String blogLink) {
        this.blogLink = blogLink;
    }

    public String getIconName() {
        return iconName;
    }

    public void setIconName(String iconName) {
        this.iconName = iconName;
    }

    public String getAvatarName() {
        return avatarName;
    }

    public void setAvatarName(String avatarName) {
        this.avatarName = avatarName;
    }

}

File: sdk/src/main/kotlin/com/cnblogs/sdk/model/UserInfo.kt
package com.cnblogs.sdk.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

/**
 *
 * 用户信息
 * @author RAE
 * @date 2022/06/21
 * @copyright Copyright (c) https://github.com/raedev All rights reserved.
 */
@Parcelize
data class UserInfo(
    /**
     * 账号
     */
    var spaceUserId: String?,

    /**
     * 昵称
     */
    var displayName: String?,

    /**
     * blogApp,非常重要的字段,唯一ID
     */
    var blogApp: String?,

    /**
     * 未读消息数量
     */
    var unreadMsg: Int = 0,

    /**
     * 个人中心主页
     */
    var homeLink: String?,

    /**
     * 博客主页
     */
    var blogLink: String?,

    /**
     * 头像小图
     */
    var iconName: String?,

    /**
     * 头像大图
     */
    var avatarName: String?
) : Parcelable
